cloudinit/sources/DataSourceAzure.py

```diff
@@ -959,7 +973,8 @@ def _check_if_nic_is_primary(self, ifname):
                 dhcp_ctx = EphemeralDHCPv4(
                     iface=ifname, dhcp_log_func=dhcp_log_cb
                 )
-                dhcp_ctx.obtain_lease()
+                lease = dhcp_ctx.obtain_lease()
```

Comment:
nit: if we're going to call process_dhcp_lease every time we obtain_lease, it might be better to add an optional callback_fn to obtain_lease and invoke that after getting the lease?

Reply:
DHCP will be consolidated in a separate PR to simplify this.